test/blackbox-tests/test-cases/pkg/withenv.t

```diff
@@ -0,0 +1,22 @@
+Setting environment variables in actions
+
+  $ mkdir dune.lock
+  $ cat >dune.lock/lock.dune <<EOF
+  > (lang package 0.1)
+  > EOF
+  $ cat >dune.lock/test <<'EOF'
```

Comment:
What's the point of the quotes around "EOF"?

Reply:
It is so that `$variables` aren't evaluated when writing the file. The lock file contains a shell snippet that prints an environment variable. We want it to be done when the shell script is executed, rather than when it is written.